File: tools/discord_push.mjs
import { Client, GatewayIntentBits } from '/root/.claude/plugins/marketplaces/claude-plugins-official/external_plugins/discord/node_modules/discord.js/src/index.js';

const GUANKE_TOKEN    = 'MTUxNTY2MjEyNDk2NDM4MDgzMw.GWoH-Z.Y_XUUQaT8EF6utY7Lkj13bzvjHI8i51XIyu1II';
const DEFAULT_CHANNEL = '1515664908489064530';

const args = process.argv.slice(2);
const [channelId, content] = args.length === 1
  ? [DEFAULT_CHANNEL, args[0]]
  : [args[0], args.slice(1).join(' ')];

if (!content) { console.error('usage: bun discord_push.mjs [channel_id] <message>'); process.exit(1); }

const client = new Client({ intents: [GatewayIntentBits.Guilds] });

client.once('clientReady', async () => {
  try {
    const ch = await client.channels.fetch(channelId);
    const sent = await ch.send(content);
    console.log(JSON.stringify({ ok: true, id: sent.id }));
  } catch (e) {
    console.error(JSON.stringify({ ok: false, error: String(e) }));
    process.exit(1);
  } finally {
    client.destroy();
    process.exit(0);
  }
});

client.login(GUANKE_TOKEN).catch(e => {
  console.error(JSON.stringify({ ok: false, error: String(e) }));
  process.exit(1);
});

setTimeout(() => { console.error(JSON.stringify({ ok: false, error: 'timeout' })); process.exit(1); }, 15000);

File: tools/discord_push.py
#!/usr/bin/env python3.11
"""
官克bot 主动推送消息到 Discord 频道。
用法：python3.11 discord_push.py <channel_id> <message>
或：  python3.11 discord_push.py <message>  （用默认频道）
"""
import sys, json
from urllib.request import urlopen, Request
from urllib.error import HTTPError

TOKEN   = 'MTUxNTY2MjIyODY1OTg5NjM0MA.GQYaLV.Akz3n4902LYbGS-OrIjKMyWlLox7mwgFV6m58U'
DEFAULT_CHANNEL = '1515664907658596433'

def send(channel_id: str, content: str) -> dict:
    url = f'https://discord.com/api/v10/channels/{channel_id}/messages'
    body = json.dumps({'content': content}).encode()
    req = Request(url, data=body, method='POST', headers={
        'Authorization': f'Bot {TOKEN}',
        'Content-Type': 'application/json',
    })
    try:
        with urlopen(req, timeout=10) as r:
            return json.loads(r.read())
    except HTTPError as e:
        return {'error': e.code, 'body': e.read().decode()}

if __name__ == '__main__':
    args = sys.argv[1:]
    if len(args) == 0:
        print("usage: discord_push.py [channel_id] <message>")
        sys.exit(1)
    elif len(args) == 1:
        ch, msg = DEFAULT_CHANNEL, args[0]
    else:
        ch, msg = args[0], ' '.join(args[1:])
    result = send(ch, msg)
    print(json.dumps(result, ensure_ascii=False))
